plugin/pulpcore/plugin/repository.py

```diff
@@ -0,0 +1,191 @@
+        >>>     changeset = ChangeSet(importer, new_version, additions=additions,
+        >>>                      removals=removals)
+        >>>
+        >>> latest_version = RepositoryVersion.latest(repository)
+        >>> content = latest_version.content
+    """
+
+    @classmethod
+    def latest(cls, repository):
+        """
+        Get the latest RepositoryVersion on a repository
+
+        Args:
+            repository (pulpcore.plugin.repository): to get the latest version of
+
+        Returns:
+            pulpcore.plugin.repository.RepositoryVersion: The latest RepositoryVersion
+
+        """
+        with suppress(models.RepositoryVersion.DoesNotExist):
+
+            model = repository.versions.exclude(complete=False).latest()
+            return RepositoryVersion(model)
+
+    @classmethod
+    def create(cls, repository):
+        """
+        Create a new RepositoryVersion
+
+        Args:
+            repository (pulpcore.plugin.repository): to create a new version of
+
+        Returns:
+            pulpcore.plugin.repository.RepositoryVersion: The Created RepositoryVersion
+
+        Raises:
+            pulpcore.exception.UnsafeOperationError: if RepositoryVersion create() is called
+                outside of a task.
+        """
+
+        if get_current_task_id() is None:
+            raise UnsafeOperationError(_('RepositoryVersion creation must be ran inside a task'))
+        with transaction.atomic():
+            version = models.RepositoryVersion(
+                repository=repository,
+                number=repository.last_version + 1)
+            repository.last_version = version.number
+            repository.save()
+            version.save()
+            resource = models.CreatedResource(content_object=version)
+            resource.save()
+            return cls(version, resource)
+
+    def __init__(self, model, resource=None):
+        """
+        Args:
+            model (pulpcore.models.repository.RepositoryVersion)
+            resource (pulpcore.models.task.CreatedResource)
+        """
+        self._model = model
+        self._created_resource = resource
+
+    @property
+    def number(self):
+        return self._model.number
+
+    @property
+    def model(self):
+        return self._model
+
+    @property
+    def content(self):
+        """
+        QuerySet of content on this repository version
+
+        Returns:
+            QuerySet: The Content objects that are related to this version.
+        """
+        return self._model.content
+
+    def add_content(self, content):
+        """
+        Add a content unit to the repository.
+
+        Args:
+           content (pulpcore.plugin.models.Content): a content model to add
+
+        Raise:
+            pulpcore.exception.RepositoryVersionExistsError: if add_content is called on a
+                complete RepositoryVersion
+        """
+        if self._model.complete:
+            raise RepostoryVersionExistsError(self._model)
```

Comment:
I dont like RepositoryVersionExists for this. The RepositoryVersion exists either way, it is just complete or not. UnsafeOperationError seems better.